graphicsmagick/src/wand/magick.rs

```diff
@@ -1333,6 +1324,90 @@ impl<'a> MagickWand<'a> {
         self.check_status(status)?;
         Ok((width, height, x, y))
     }
+}
+
+#[derive(Debug)]
+pub struct MagickWandExportSlice<'a, T> {
+    columns: c_ulong,
+    rows: c_ulong,
+    map: &'a str,
+    slice: &'a mut [MaybeUninit<T>],
+
+    /// The number of `T` that will be written to `slice`.
+    len: usize,
+}
+
+#[allow(clippy::len_without_is_empty)]
+impl<'a, T> MagickWandExportSlice<'a, T>
+where
+    T: MagickWandExportType,
+{
+    /// Return `Some` if the `slice` is large enough.
+    /// Otherwise return `None`.
+    pub fn new(
+        columns: c_ulong,
+        rows: c_ulong,
+        map: &'a str,
+        slice: &'a mut [MaybeUninit<T>],
+    ) -> Option<Self> {
+        let size: usize = (columns * rows).try_into().unwrap();
+        let len = size * map.len();
+
+        if slice.len() >= len {
+            Some(Self {
+                columns,
+                rows,
+                map,
+                slice,
+                len,
+            })
+        } else {
+            None
+        }
+    }
+
+    /// The number of `T` that will be written to `slice`.
+    pub const fn len(&self) -> usize {
+        self.len
+    }
+}
+
+impl MagickWand<'_> {
+    /// Check the doc and implementation of [`MagickWand::get_image_pixels`].
+    pub fn write_image_pixels_to<'a, T: MagickWandExportType>(
```

Comment:
I would like this function to be public because it enables user to get image pixels into their existing buffer without allocating an intermediate `Vec`.

My use case here is to port [`Image:toTensor`](https://github.com/clementfarabet/graphicsmagick/blob/203df4457613fa612fabe3180217b48fff1d802b/Image.lua#L958) and I might need this function for best performance.